tests/test_views.py

```diff
@@ -0,0 +1,1050 @@
+        """Test the context variables which are similar across different paths.
+
+        These context variables do not change based on the specific path taken
+        in the view, and so are being tested apart from the rest. The only time
+        they may change is if the requested url does not actually exist, and that
+        occurrence is covered in the previous test.
+        """
+        entity = u'http://www.example.com'
+        entity_surt = u'http://(com,example,www,)'
+        project = factories.ProjectFactory()
+        project_metadata = factories.ProjectMetadataFactory(project=project)
+        url = factories.URLFactory(url_project=project, entity=entity)
+        factories.SURTFactory(
+            url_project=project,
+            entity=entity,
+            value=entity_surt
+        )
+        related_urls = factories.SURTFactory.create_batch(
+            1,
+            url_project=project,
+            entity=u'{}/main/page'.format(entity),
+            value=u'{}/main/page'.format(entity_surt)
+        )
+        expected_context = {
+            'project': project,
+            'url_data': views.create_url_list(project, related_urls + [url]),
+            'related_url_list': related_urls,
+            'metadata_vals': views.get_metadata(project),
+            'institutions': views.get_look_ahead(project),
+            'form_types': json.dumps({project_metadata.metadata.name: project_metadata.form_type}),
+        }
+        response = client.get(reverse('url_listing', args=[project.project_slug, entity]))
+
+        for key in expected_context:
+            assert str(response.context[key]) == str(expected_context[key])
+
+    def test_context_with_post(self, client):
+        data = {
+            'scope': 1,
+            'nominator_name': 'John Doe',
+            'nominator_email': 'here@there.com',
+            'nominator_institution': 'UNT',
+            'metadata': '2012-12-12'
+        }
+        entity = 'www.example.com'
+        project = factories.ProjectFactory(registration_required=False)
+        factories.ProjectMetadataFactory(
+            project=project,
+            metadata__name='metadata',
+            form_type='date'
+        )
+        factories.URLFactory(url_project=project, entity=entity)
+        response = client.post(
+            reverse('url_listing', args=[project.project_slug, entity]),
+            data
+        )
+        expected = {
+            'scope_form': views.ScopeForm(data),
+            'form_errors': '',
+            'summary_list': [
+                u'You have successfully nominated www.example.com',
+                u'You have successfully added the metadata "2012-12-12" for www.example.com'
+            ],
+            'json_data': None,
+        }
+
+        for key in expected:
+            assert str(response.context[key]) == str(expected[key])
+
+    @pytest.mark.parametrize('reg_required, data, expected_errors', [
```

Comment:
Is it the size of the parametrization that concerns you or the number of parameters, or both?